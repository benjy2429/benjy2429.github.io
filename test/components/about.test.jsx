import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import About from '../../src/components/about';

const defaultProps = [
  {
    node: {
      html: 'foo',
      frontmatter: {
        title: 'Work'
      }
    }
  },
  {
    node: {
      html: 'bar',
      frontmatter: {
        title: 'Education'
      }
    }
  }
];

describe('About component', () => {
  it('renders correctly', () => {
    const component = shallow(<About data={defaultProps} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders correctly when data is not available', () => {
    const component = shallow(<About data={[]} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
