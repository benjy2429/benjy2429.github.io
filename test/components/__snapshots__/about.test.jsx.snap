// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component renders correctly 1`] = `
<section
  className="about container"
>
  <div
    className="about-wrapper"
  >
    <div
      className="about-item"
    >
      <h2
        id="work"
      >
        Work
      </h2>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "foo",
          }
        }
      />
    </div>
    <div
      className="about-item"
    >
      <h2
        id="education"
      >
        Education
      </h2>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "bar",
          }
        }
      />
    </div>
  </div>
</section>
`;

exports[`About component renders correctly when data is not available 1`] = `
<section
  className="about container"
>
  <div
    className="about-wrapper"
  />
</section>
`;
